Code/GraphMol/FileParsers/MaeMolSupplier.cpp

```diff
@@ -10,243 +10,424 @@
+      continue;
+    } else if (!prop.second->isDefined(i)) {
+      continue;
     }
 
-    if(pdb_info.m_resnum && pdb_info.m_resnum->isDefined(atom_num))
-    {
-        rd_info->setResidueNumber(pdb_info.m_resnum->at(atom_num));
+    atom.setProp(prop.first, prop.second->at(i));
+  }
+
+  for (const auto &prop : atom_block.getProperties<double>()) {
+    if (prop.first == mae::ATOM_X_COORD || prop.first == mae::ATOM_Y_COORD ||
+        // Coordinates are used in defining a conformation, and should not be
+        // set on the atom.
+        prop.first == mae::ATOM_Z_COORD) {
+      continue;
+    } else if (prop.first == PDB_OCCUPANCY || prop.first == PDB_TFACTOR) {
+      // PDB information is parsed separately.
+      continue;
+    } else if (!prop.second->isDefined(i)) {
+      continue;
     }
 
-    if(pdb_info.m_occupancy && pdb_info.m_occupancy->isDefined(atom_num))
-    {
-        rd_info->setOccupancy(pdb_info.m_occupancy->at(atom_num));
+    atom.setProp(prop.first, prop.second->at(i));
+  }
+  for (const auto &prop : atom_block.getProperties<int>()) {
+    if (prop.first == mae::ATOM_ATOMIC_NUM) {
+      // Atomic number was already used in the creation of the atom
+      continue;
+    } else if (prop.first == PDB_RESIDUE_NUMBER) {
+      // PDB information is parsed separately.
+      continue;
+    } else if (!prop.second->isDefined(i)) {
+      continue;
     }
 
-    if(pdb_info.m_tempfac && pdb_info.m_tempfac->isDefined(atom_num))
-    {
-        rd_info->setTempFactor(pdb_info.m_tempfac->at(atom_num));
+    if (prop.first == mae::ATOM_FORMAL_CHARGE) {
+      // Formal charge has a specific setter
+      atom.setFormalCharge(prop.second->at(i));
+    } else {
+      atom.setProp(prop.first, prop.second->at(i));
+    }
+  }
+  for (const auto &prop : atom_block.getProperties<mae::BoolProperty>()) {
+    if (!prop.second->isDefined(i)) {
+      continue;
     }
+
+    atom.setProp(prop.first, static_cast<bool>(prop.second->at(i)));
+  }
 }
 
-void addAtoms(shared_ptr<Block>& current_struct, RWMol& mol)
-{
-  // Atom data is in the m_atom indexed block
-  {
-    const auto atom_data = current_struct->getIndexedBlock("m_atom");
-    // All atoms are guaranteed to have these three field names:
-    const auto atomic_numbers = atom_data->getIntProperty("i_m_atomic_number");
-    const auto xs = atom_data->getRealProperty("r_m_x_coord");
-    const auto ys = atom_data->getRealProperty("r_m_y_coord");
-    const auto zs = atom_data->getRealProperty("r_m_z_coord");
-    const auto atom_count = atomic_numbers->size();
-    shared_ptr<IndexedIntProperty> atomic_charges;
-    try {
-      atomic_charges = atom_data->getIntProperty("i_m_formal_charge");
-    } catch (std::out_of_range &) {
-    }
+void addAtoms(const mae::IndexedBlock &atom_block, RWMol &mol) {
+  // All atoms are guaranteed to have these three field names:
+  const auto atomic_numbers = atom_block.getIntProperty(mae::ATOM_ATOMIC_NUM);
```

Comment:
not quite true. All atoms have either atomic number or atom type (or both).

Reply:
Yeah, but we don't have a way of translating to/from atom types without Schrodinger libraries, so we can't handle atom types, which means that need atomic numbers.